pkg/dockerregistry/server/auth.go

```diff
@@ -535,3 +545,11 @@ func verifyPruneAccess(ctx context.Context, client client.SubjectAccessReviews)
 	}
 	return nil
 }
+
+func isMetricsBearerToken(ctx context.Context, token string) bool {
+	config := ConfigurationFrom(ctx)
+	if config.Metrics.Enabled {
+		return config.Metrics.Secret == token
+	}
```

Comment:
I don't agree with that.  If you are already doing auth via the master, we want metrics to use that behavior.  Static secret is only for things that aren't integrated with master auth (which is a limited set, but registry is under)